config/v1/types_infrastructure.go

```diff
@@ -1020,6 +1020,8 @@ type PowerVSPlatformSpec struct {
 }
 
 // PowerVSPlatformStatus holds the current status of the IBM Power Systems Virtual Servers infrastrucutre provider.
+// +kubebuilder:validation:XValidation:rule="oldSelf.resourceGroup == '' || self.resourceGroup == oldSelf.resourceGroup",message="resourceGroup is immutable once set"
```

Comment:
You can move this one to the field directly, I think that would be preferred please
```suggestion
// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="resourceGroup is immutable once set"
```

Reply:
Ok moved to field itself.